tests/unit/runtime/test_multiple_models.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import pytest
+import deepspeed
+import deepspeed.comm as dist
+import torch
+from unit.common import DistributedTest
+from unit.simple_model import SimpleModel, random_dataloader
+
+
+def create_model(config_dict):
+    hidden_dim = 64
+    model = SimpleModel(hidden_dim)
+    model, _, _, _ = deepspeed.initialize(model=model, model_parameters=model.parameters(), config=config_dict)
+    return model
+
+
+def train_loop(num_models, config_dict, dtype):
+    hidden_dim = 64
+
+    models = [create_model(config_dict) for _ in range(num_models)]
+    data_loader = random_dataloader(model=models[0],
+                                    total_samples=4,
+                                    hidden_dim=hidden_dim,
+                                    device=models[0].device,
+                                    dtype=dtype)
+    dist.barrier()
+    for _, batch in enumerate(data_loader):
+        losses = [m.module(batch[0], batch[1]) for m in models]
+        loss = sum(l / (i + 1) for i, l in enumerate(losses))
+        loss.backward()
```

Comment:
should the case with each model driving its own `backward` be tested as well? 

I'm not sure independent models will want to be connected and impact each other via autograd as is the case here.

probably could be done in this same test as another loop?

Reply:
Can you clarify a bit more, because I think the case of each model driving its own backward is different from this case of composing individual losses into a `global` loss.  

More specifically, are you referring to a case like below? 
```python
    models = [...]
    for batch in dataset:
       losses = [m(batch) for m in models]
       for m, loss in zip(models, losses):
          m.backward(loss)
  ```

Reply:
yes, that would be one of them. I surely can't think of all use-cases, but there will be definitely some that will want to independent engines, running at the same time.